### PR TITLE
feat(Esm): use `gen-esm-wrapper` instead of manually making the file

### PR DESCRIPTION
**Pull request by vladfrangu**
**Please describe the changes this PR makes and why it should be merged:**

One less file to maintain! 🎉 

**Status and versioning classification:**

N/A?

### CODE REVIEW

**Review by iCrawl — APPROVED** [review]

**Review by kyranet — APPROVED** [review]

**Review by SpaceEEC — APPROVED** [review]

**Comment by kyranet** [comment]
@vladfrangu this needs a rebase.

File: .gitignore [hunk 1]
```diff
@@ -20,3 +20,5 @@ deploy/deploy_key.pub
 .idea/
 docs/docs.json
 typings/index.js
+# Autogenerated
```
Comment (vaporox) [inline .gitignore:23]: Smallest nitpick ever
```suggestion

# Autogenerated
```

File: package.json [hunk 2]
```diff
@@ -3,16 +3,17 @@
...
       {
         "require": "./src/index.js",
-        "import": "./esm/discord.mjs"
+        "import": "./src/index.mjs"
       },
       "./src/index.js"
     ],
-    "./esm": "./esm/discord.mjs"
+    "./esm": "./src/index.mjs"
   },
```
Comment (vaporox) [inline package.json:17]: Couldn't the whole exports object be flattened?
```json
"exports": {
  "import": "./src/index.mjs",
  "require": "./src/index.js"
},
```

**Comment by Litleck** [comment]
I don't really understand this too much, but I decided to switch to TypeScript and spent over an hour trying to find why it wouldn't work. Eventually I saw that it had no mjs file and it led me to this PR, then manually running `gen-esm-wrapper ./src/index.js ./src/index.mjs` fixed it. I don't know if this is supposed to automatically run sometime or I'm misunderstanding

**Comment by iShibi** [comment]
I runs automatically when the library gets packaged and published to npm. So, you wouldn't have to do this if you install from npm using `npm i discord.js@dev`

**Comment by monbrey** [comment]
>I decided to switch to TypeScript and spent over an hour trying to find why it wouldn't work. Eventually I saw that it had no mjs file

TypeScript doesn't use the .mjs file, it's only for using ES6 syntax in regular JavaScript

**Pull request merged**